File: codeforces/1000/beautiful_matrix.py
#!/usr/bin/env pypy
# https://codeforces.com/problemset/problem/263/A
coord_x, coord_y = 0, 0
for line_number in range(5):
    line = input()
    if not "1" in line:
        continue
    coord_y = line_number + 1
    coord_x = line.split().index("1") + 1
    break
print(abs(3 - coord_x) + abs(3 - coord_y))

File: codeforces/1000/beautiful_matrix.pas
program BeautifulMatrix;

var
  coord_x, coord_y, line_number: Integer;
  line: String;
begin
  coord_x := 0;
  coord_y := 0;
  for line_number := 1 to 5 do
  begin
    Readln(line);
    if Pos('1', line) = 0 then
      Continue;
    coord_y := line_number;
    coord_x := ((Pos('1', line) - 1) div 2) + 1;
    Break;
  end;
  WriteLn(Abs(3 - coord_x) + Abs(3 - coord_y));
end.

